File: apps/dsa-web/src/types/xinfa.ts
export interface XinfaEntryItem {
  id: number;
  title: string;
  content: string;
  category: string;
  tags: string[] | null;
  stockCode: string | null;
  stockName: string | null;
  sentiment: number | null;
  isStarred: boolean;
  createdAt: string | null;
  updatedAt: string | null;
}

export interface XinfaListResponse {
  total: number;
  items: XinfaEntryItem[];
  page: number;
  pageSize: number;
}

export interface XinfaCategorySummaryItem {
  category: string;
  count: number;
  label: string;
}

export interface XinfaCategorySummaryResponse {
  categories: XinfaCategorySummaryItem[];
}

export interface XinfaCreateRequest {
  title: string;
  content: string;
  category?: string;
  tags?: string[];
  stockCode?: string;
  stockName?: string;
  sentiment?: number;
  isStarred?: boolean;
}

export interface XinfaUpdateRequest {
  title?: string;
  content?: string;
  category?: string;
  tags?: string[];
  stockCode?: string;
  stockName?: string;
  sentiment?: number;
  isStarred?: boolean;
}

export const CATEGORY_LABELS: Record<string, string> = {
  general: '通用',
  review: '复盘反思',
  discipline: '交易纪律',
  mindset: '心态建设',
  experience: '经验总结',
  plan: '交易计划',
};

export const CATEGORY_LIST = Object.entries(CATEGORY_LABELS).map(([key, label]) => ({
  key,
  label,
}));

// ---------------------------------------------------------------------------
// 树干树枝 (Knowledge Tree) 数据类型
// ---------------------------------------------------------------------------

export interface TreeNode {
  id: string;
  label: string;
  description: string;
  children: TreeNode[];
  collapsed: boolean;
}

export interface KnowledgeTree {
  id: string;
  title: string;
  nodes: TreeNode[]; // 根节点 = 树干
  createdAt: string;
  updatedAt: string;
}

export function createNodeId(): string {
  return `node_${Date.now()}_${Math.random().toString(36).slice(2, 8)}`;
}

export function createEmptyTree(title?: string): KnowledgeTree {
  return {
    id: createNodeId(),
    title: title ?? '新建知识树',
    nodes: [],
    createdAt: new Date().toISOString(),
    updatedAt: new Date().toISOString(),
  };
}

export function createEmptyNode(label?: string): TreeNode {
  return {
    id: createNodeId(),
    label: label ?? '新节点',
    description: '',
    children: [],
    collapsed: false,
  };
}

File: api/v1/schemas/xinfa.py
# -*- coding: utf-8 -*-
"""
心法模块 API Schema
"""

from typing import List, Optional
from pydantic import BaseModel, Field


# ---------------------------------------------------------------------------
# 请求
# ---------------------------------------------------------------------------

class XinfaCreateRequest(BaseModel):
    title: str = Field(..., min_length=1, max_length=200, description="标题")
    content: str = Field(..., min_length=1, description="正文（Markdown）")
    category: str = Field("general", description="分类: general/review/discipline/mindset/experience/plan")
    tags: Optional[List[str]] = Field(None, description="标签列表")
    stock_code: Optional[str] = Field(None, max_length=10, description="关联股票代码")
    stock_name: Optional[str] = Field(None, max_length=50, description="关联股票名称")
    sentiment: Optional[int] = Field(None, ge=1, le=5, description="情绪评分 1-5")
    is_starred: bool = Field(False, description="是否标星")


class XinfaUpdateRequest(BaseModel):
    title: Optional[str] = Field(None, min_length=1, max_length=200)
    content: Optional[str] = Field(None, min_length=1)
    category: Optional[str] = None
    tags: Optional[List[str]] = None
    stock_code: Optional[str] = None
    stock_name: Optional[str] = None
    sentiment: Optional[int] = Field(None, ge=1, le=5)
    is_starred: Optional[bool] = None


class XinfaListParams(BaseModel):
    category: Optional[str] = Field(None, description="分类筛选")
    keyword: Optional[str] = Field(None, description="关键词搜索（标题/内容）")
    stock_code: Optional[str] = Field(None, max_length=10, description="关联股票代码筛选")
    is_starred: Optional[bool] = Field(None, description="仅标星")
    page: int = Field(1, ge=1, description="页码")
    page_size: int = Field(20, ge=1, le=100, description="每页条数")


# ---------------------------------------------------------------------------
# 响应
# ---------------------------------------------------------------------------

class XinfaEntryItem(BaseModel):
    id: int = Field(..., description="条目ID")
    title: str = Field(..., description="标题")
    content: str = Field(..., description="正文")
    category: str = Field("general", description="分类")
    tags: Optional[List[str]] = Field(None, description="标签列表")
    stock_code: Optional[str] = Field(None, description="关联股票代码")
    stock_name: Optional[str] = Field(None, description="关联股票名称")
    sentiment: Optional[int] = Field(None, description="情绪评分")
    is_starred: bool = Field(False, description="是否标星")
    created_at: Optional[str] = Field(None, description="创建时间")
    updated_at: Optional[str] = Field(None, description="更新时间")


class XinfaListResponse(BaseModel):
    total: int = Field(0, description="总数")
    items: List[XinfaEntryItem] = Field(default_factory=list, description="条目列表")
    page: int = Field(1, description="当前页码")
    page_size: int = Field(20, description="每页条数")


class XinfaCategorySummaryItem(BaseModel):
    category: str = Field(..., description="分类代码")
    count: int = Field(0, description="条目数")
    label: str = Field("", description="中文分类名")


class XinfaCategorySummaryResponse(BaseModel):
    categories: List[XinfaCategorySummaryItem] = Field(default_factory=list)


class XinfaDeleteResponse(BaseModel):
    success: bool = Field(..., description="是否成功")
